### PR TITLE
feat: Initial generation of google-cloud-filestore-v1

### PR DESCRIPTION
**Pull request by dazuma**
Auto-created at 2021-08-03 01:35:23 +0000 using `toys gen-client`.

### CODE REVIEW

File: google-cloud-filestore-v1/.repo-metadata.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "api_id": "file.googleapis.com",
+    "client_documentation": "https://googleapis.dev/ruby/google-cloud-filestore-v1/latest",
+    "distribution_name": "google-cloud-filestore-v1",
+    "language": "ruby",
+    "name": "file",
+    "name_pretty": "Filestore V1 API",
+    "product_documentation": "https://cloud.google.com/filestore/",
+    "repo": "googleapis/google-cloud-ruby",
+    "requires_billing": true,
```
Comment (quartzmo) [inline google-cloud-filestore-v1/.repo-metadata.json:10]: Is it OK that this library doesn't have `ruby-cloud-env-prefix`?

File: google-cloud-filestore-v1/lib/google/cloud/filestore/v1/cloud_filestore_manager.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        # `region` for backups; for example:
+        # * `projects/12345/locations/us-central1-c/instances/my-filestore`
+        # * `projects/12345/locations/us-central1/backups/my-backup`
+        #
+        # To load this service and instantiate a client:
+        #
+        #     require "google/cloud/filestore/v1/cloud_filestore_manager"
+        #     client = ::Google::Cloud::Filestore::V1::CloudFilestoreManager::Client.new
+        #
+        module CloudFilestoreManager
```
Comment (quartzmo) [inline google-cloud-filestore-v1/lib/google/cloud/filestore/v1/cloud_filestore_manager.rb:60]: Did you want to use `ruby-cloud-namespace-override` to remove the atypical `Cloud` from `CloudFilestoreManager`?

File: google-cloud-filestore-v1/.repo-metadata.json (hunk 1 above)
Comment (dazuma) [reply google-cloud-filestore-v1/.repo-metadata.json:10]: I've now stopped adding the prefix for all new libraries. We're fast polluting the environment variable namespace, and increasing the possibility that one of the environment variable names we read will collide with a user's application env variable, resulting in us reading/using the wrong credentials. I think it was a mistake to provide that prefix for any libraries beyond our often-used veneers.

File: google-cloud-filestore-v1/lib/google/cloud/filestore/v1/cloud_filestore_manager.rb (hunk 2 above)
Comment (dazuma) [reply google-cloud-filestore-v1/lib/google/cloud/filestore/v1/cloud_filestore_manager.rb:60]: No, we have a bunch of cases of that kind of naming in other libraries, and I don't think the aesthetic improvement is worth the risks of changing the name from what the proto specifies. The only time we use namespace overrides are to fix capitalization edge cases (like AutoML) or to remove version-specific suffixes.

**Pull request merged**